R/coord-sf.R

```diff
@@ -582,6 +591,50 @@ parse_axes_labeling <- function(x) {
   list(top = labs[1], right = labs[2], bottom = labs[3], left = labs[4])
 }
 
+# This function does two things differently from standard breaks:
+#   1. It does not resolve `waiver()`, unless `n.breaks` is given. In the case
+#      that breaks are `waiver()`, we use the default graticule breaks.
+#   2. It discards non-finite breaks because they are invalid input to the
+#      graticule. This may cause atomic `labels` to be out-of-sync.
+sf_breaks <- function(scale_x, scale_y, bbox, crs) {
+
+  x_breaks <- y_breaks <- waiver()
+
+  has_x <- !is.null(scale_x$breaks) || !is.null(scale_x$n.breaks)
+  has_y <- !is.null(scale_y$breaks) || !is.null(scale_y$n.breaks)
+
+
+  if (has_x || has_y) {
+    if (!is.null(crs)) {
+      # Atomic breaks input are assumed to be in long/lat coordinates.
+      # To preserve that assumption for function breaks, the bounding box
+      # needs to be translated to long/lat coordinates.
+      if (!is_named(bbox)) {
+        names(bbox) <- c("xmin", "ymin", "xmax", "ymax")
+      }
+      # Convert bounding box to long/lat coordinates
+      bbox <- sf::st_as_sfc(sf::st_bbox(bbox, crs = crs))
+      bbox <- sf::st_bbox(sf::st_transform(bbox, 4326))
+      bbox <- as.numeric(bbox)
+
+      # If any bbox is NA the transformation has probably failed.
+      # (.e.g from IGH to long/lat). In this case, just provide full long/lat.
+      bbox[is.na(bbox)] <- c(-180, -90, 180, 90)[is.na(bbox)]
+    }
```

Comment:
I'd really like some more sf-experienced eyes on this bit

Reply:
@clauswilke you are perhaps the one best fitting the "sf-experienced" moniker :-)

Reply:
I haven't looked carefully yet and I'm not entirely sure I even understand what this code is meant to do, but one thing I notice right away is hard-coded CRS values (line 617) and limits (line 622). Should this respect the choice of `default_crs` in `coord_sf()`? Also, I'm worried about just transforming a bounding box. This can give all sorts of unexpected results for weird coordinate systems. That's why I added different limits methods implemented in `calc_limits_bbox()`. Should that be used here, and the choice of the limit method respected (see argument `lims_method` in `coord_sf()`)? Finally, I added the function `sf_transform_xy()` to transform coordinates will imposing various checks. Is that useful here? (The answer to several of my questions may well be "no", just wanted to be sure I raised them so we've thought it all through carefully.)